### ISSUE TITLE
AOTAutograd based torch.compile doesn't capture manual seed setting in the graph

### ISSUE THREAD

**Issue by sujoysaraswati** [pytorch/pytorch#95161]
### 🐛 Describe the bug

The following code breaks with AOTautograd based 2.0 torch.compile -
```
from typing import List
import torch
import torch._dynamo as dynamo
from torch._functorch.aot_autograd import aot_module_simplified

dynamo.reset()

def my_non_aot_compiler(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]):
    print(gm.code)
    return gm.forward

def my_aot_compiler(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]):
    def my_compiler(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]):
        print(gm.code)
        return gm.forward

    # Invoke AOTAutograd
    return aot_module_simplified(
        gm,
        example_inputs,
        fw_compiler=my_compiler
    )

def my_example():
    torch.manual_seed(0)
    d_float32 = torch.rand((8, 8), device="cpu")
    return d_float32 + d_float32

compiled_fn = torch.compile(backend=my_aot_compiler)(my_example)
#compiled_fn = torch.compile(backend=my_non_aot_compiler)(my_example)

r1 = compiled_fn()
r2 = compiled_fn()
print("Results match? ", torch.allclose(r1, r2, atol = 0.001, rtol = 0.001))
```
When ```my_aot_compiler``` is used, the result is wrong as the graph doesn't capture torch.manual_seed -
```
def forward(self):
    rand = torch.ops.aten.rand.default([8, 8], device = device(type='cpu'), pin_memory = False)
    add = torch.ops.aten.add.Tensor(rand, rand);  rand = None
    return (add,)
    
Results match?  False
/usr/local/lib/python3.8/dist-packages/torch/_functorch/aot_autograd.py:1251: UserWarning: Your compiler for AOTAutograd is returning a a function that doesn't take boxed arguments. Please wrap it with functorch.compile.make_boxed_func or handle the boxed arguments yourself. See https://github.com/pytorch/pytorch/pull/83137#issuecomment-1211320670 for rationale.
  warnings.warn(
```

However, when a non-AOT autograd based compiler ```my_non_aot_compiler``` is used, torch.manual_seed is captured in the graph and the result is correct -
```
def forward(self):
    manual_seed = torch.random.manual_seed(0)
    rand = torch.rand((8, 8), device = 'cpu')
    add = rand + rand;  rand = None
    return (add,)
    
Results match?  True
```

### Versions

PyTorch version: 2.0.0.dev20230220+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: 10.0.0-4ubuntu1 
CMake version: version 3.25.2
Libc version: glibc-2.31

Python version: 3.8.10 (default, Nov 14 2022, 12:59:47)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.10.147+-x86_64-with-glibc2.29
Is CUDA available: True
CUDA runtime version: 11.6.124
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: Tesla T4
Nvidia driver version: 510.47.03
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.4.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   46 bits physical, 48 bits virtual
CPU(s):                          2
On-line CPU(s) list:             0,1
Thread(s) per core:              2
Core(s) per socket:              1
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           85
Model name:                      Intel(R) Xeon(R) CPU @ 2.00GHz
Stepping:                        3
CPU MHz:                         2000.140
BogoMIPS:                        4000.28
Hypervisor vendor:               KVM
Virtualization type:             full
L1d cache:                       32 KiB
L1i cache:                       32 KiB
L2 cache:                        1 MiB
L3 cache:                        38.5 MiB
NUMA node0 CPU(s):               0,1
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Mitigation; PTE Inversion
Vulnerability Mds:               Vulnerable; SMT Host state unknown
Vulnerability Meltdown:          Vulnerable
Vulnerability Mmio stale data:   Vulnerable
Vulnerability Retbleed:          Vulnerable
Vulnerability Spec store bypass: Vulnerable
Vulnerability Spectre v1:        Vulnerable: __user pointer sanitization and usercopy barriers only; no swapgs barriers
Vulnerability Spectre v2:        Vulnerable, IBPB: disabled, STIBP: disabled, PBRSB-eIBRS: Not affected
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Vulnerable
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single ssbd ibrs ibpb stibp fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm mpx avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves arat md_clear arch_capabilities

Versions of relevant libraries:
[pip3] numpy==1.24.2
[pip3] pytorch-triton==2.0.0+c8bfe3f548
[pip3] torch==2.0.0.dev20230220+cu117
[pip3] torchaudio==0.13.1+cu116
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.14.1
[pip3] torchvision==0.14.1+cu116
[conda] Could not collect

**Comment by ezyang** [comment]
Yeah, this is an AOTAutograd limitation. A stop gap for soundness would be to force a graph break on seeding. This is similar to https://github.com/pytorch/pytorch/issues/92804

**Comment by bdhirsh** [comment]
Adding the dynamo label since the short-term solution is to graph break

**Comment by williamwen42** [comment]
Dynamo is currently graph breaking on the `manual_seed`. Assigning to back to AOTAutograd for a more long-term solution.
